client/src/components/Balance/BalanceTransfer.jsx

```diff
@@ -1,14 +1,107 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react';
+import CurrentBalance from './CurrentBalance.jsx';
+import BalanceAPIutils from './BalanceAPIutils'
 
-function BalanceTransfer() {
+function BalanceTransfer({ userId }) {
+  const [ currentBalance, setCurrentBalance ] = useState(0);
+  const [ amount, setAmount ] = useState('');
+  const [ driverEmail, setDriverEmail ] = useState('');
+  const [ displayBalanceError, setDisplayBalanceError ] = useState(false);
+  const [ displaySelfError, setDisplaySelfError ] = useState(false);
+  const [ displayNotFoundError, setDisplayNotFoundError ] = useState(false);
+  const [ displaySuccessMessage, setDisplaySuccessMessage ] = useState(false);
+  const [ displayTipForm, setDisplayTipForm ] = useState(true);
+
+
+
+  const getUserBalance = async () => {
+      const { data } = await BalanceAPIutils.getBalance(userId)
+      setCurrentBalance(data);
+  }
+
+  const handleSend = async () => {
+    try {
+      setDisplaySuccessMessage(false)
+      setDisplaySelfError(false);
+      setDisplayNotFoundError(false);
+      if (amount > currentBalance) {
+        setDisplayBalanceError(true)
+      } else {
+        setDisplayBalanceError(false)
+        await BalanceAPIutils.transfer(userId, driverEmail, amount)
+        setDisplaySuccessMessage(true)
+        setDisplayTipForm(false)
+        getUserBalance()
+      }
+    } catch (err) {
+      if (err.response.status === 405) {
+        setDisplaySelfError(true)
+      }
+      if (err.response.status === 404) {
+        setDisplayNotFoundError(true)
+      }
+    }
+  }
+
```

Comment:
I'm not family with useState and useEffect, look like we don't need to do this.setState if we use useState